iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -317,10 +344,36 @@ static void pullInProducersInSameGroup(
             rewriter, clonedOpToFuse->getResult(opResult.getResultNumber()),
             tiledOp.getShapedOpOperand(en.index()));
         if (!maybeFusionInfo.hasValue()) {
+          LLVM_DEBUG(llvm::dbgs() << "failed to fuse with tensor\n");
           rewriter.replaceOp(clonedOpToFuse, producer->getResults());
         } else {
+          LLVM_DEBUG(llvm::dbgs() << "succeeded to fuse with tensor\n");
           maybeFusionInfo->fusedProducer.getOperation()->removeAttr(
               kFusionGroupsAttr);
+          fusedProducer = maybeFusionInfo->fusedProducer;
+        }
+      }
+
+      // If the producer is successfully fused, go recursive over the current
+      // producer's operands and pull them in if they are marked to be fused
+      // into the current group.
+      if (fusedProducer) {
```

Comment:
Maybe leave a TODO here. THis is strange layering coming from the fact that this method is trying to see if it can pull in all its producers. It should actually just be trying to pull in one producer at a time. Then we dont need these two `if` conditions.

Reply:
Added TODO. Also removed the `if` condidtions here: we have the check in the recursive call anyway. Having it here avoids ever going into recursion if not necessary; but the gain is marginal and it complicates the logic.